src/process/process_set.cu

```diff
@@ -4,71 +4,92 @@
 
 #include <chrono>
 #include <iostream>
+#include <micm/util/cuda_param.hpp>
+
+//device pointers passing to AddForcingTermsKernel()
+typedef struct forcingDevice{
+  double* rate_constants; 
+  double* state_variables; 
+  double* forcing; 
+  size_t* number_of_reactants; 
+  size_t* reactant_ids; 
+  size_t* number_of_products; 
+  size_t* product_ids; 
+  double* yields; 
+};
+//device pointers passing to AddJacobianTermsKernel() 
+typedef struct jacobianDevice{
+  double* rate_constants; 
+  double* state_variables; 
+  double* jacobian;
+  size_t* number_of_reactants; 
+  size_t* reactant_ids; 
+  size_t* number_of_products; 
+  double* yields; 
+  size_t* jacobian_flat_ids; 
+};
+
+const size_t BLOCK_SIZE = 320;
 
 namespace micm
 {
   namespace cuda
   {
     // flipped memory layout
     __global__ void AddForcingTermsKernel(
-        double* rate_constants,
-        double* state_variables,
-        double* forcing,
+        forcingDevice* device, 
         size_t n_grids,
         size_t n_reactions,
-        size_t n_species,
-        size_t* number_of_reactants_,
-        size_t* reactant_ids_,
-        size_t* number_of_products_,
-        size_t* product_ids_,
-        double* yields_)
+        size_t n_species)
```

Comment:
would it make sense to also include these in the `ForcingDevice` struct?

Reply:
My initial thought was to only group the vectors. It does make sense to include these variables. Changes made.